server/app/middlewares/websocket_backend.rb

```diff
@@ -335,25 +369,33 @@ def on_pong(client, delay)
     end
 
     node = HostNode.find_by(node_id: client[:id])
+
     if node
-      if node.connected?
-        node.set(last_seen_at: Time.now.utc)
-      else
-        logger.warn "Close connection of disconnected node #{node.name || node.node_id}"
-        close_client(client)
+      connected_node = HostNode.where(id: node.id, connected_at: client[:connected_at])
+        .find_one_and_update({:$set => {last_seen_at: Time.now.utc}})
+
+      if !connected_node
+        logger.warn "Close conflicting connection for node #{node} connected at #{client[:connected_at]}, node has re-connected at #{node.connected_at}"
+        close_client(client, 4041, "host node #{node} connection conflict with new connection at #{node.connected_at}")
```

Comment:
This check should trigger if there are multiple active connections with the same Docker/Node ID, such as for cloned VMs. Tested this by running two agents on the same host node.

How those agents will then behave is rather odd, it depends on whether or not they connect to the same server puma worker or not... If they both connect to the same worker, then both clients end up in the same `WebsocketBackend@clients` array, and the `client_for_id` used to send RPCs to the agents will pick the first one to have connected... so after a reconnect, they don't get their `master_info` or `node_info` RPCs, those go to the conflicting agent.